docs/userguide/CDN/USERGUIDE.md

```diff
@@ -0,0 +1,289 @@
+# Complete User Guide for the CDN Service
+
+CDN is a service that you can use to manage your CDN-enabled domains and the origins and assets associated with those domains.
+
+## Table of contents
+  * [Concepts](#concepts)
+  * [Prerequisites](#prerequisites)
+    * [Client](#client)
+    * [CDN service](#cdn-service)
+  * [Services](#services)
+    * [Create a service](#create-a-service-to-represent-your-web-application)
+    * [List Services](#list-services)
+    * [Get a service](#get-a-service)
+    * [Update a service](#update-a-service)
+    * [Delete a service](#delete-a-service)
+  * [Service Assets](#service-assets)
+    * [Purge all cached service assets](#purge-all-cached-service-assets)
+    * [Purge a specific cached service asset](#purge-a-specific-cached-service-asset)
+  * [Flavors](#flavors)
+    * [Create a flavor](#create-a-flavor)
+    * [List flavors](#list-flavors)
+    * [Get a flavor](#get-a-flavor)
+    * [Delete a flavor](#delete-a-flavor)
+
+## Concepts
+
+To use the CDN service effectively, you should understand the following
+key concepts:
+
+* **Content delivery network**: A content delivery network (CDN) is a system of multiple computers that contains copies of data stored at various network nodes. A CDN is designed to improve performance of publicly distributed assets. Assets can be anything from website content, to web application components, to media such as videos, ads, and interactive experiences.  CDNs decrease the load time of these assets by caching them on edge servers, also called points of presence (PoPs).  Edge servers are distributed around the globe, meaning requests only travel to a local location to grab assets, rather than to and from a data center based far from the end user.
+
+* **Edge node**: CDN providers have many points of presence (PoP) servers around the world. These servers are known as edge nodes. These edge nodes cache the content and serve it directly to customers, thus reducing transit time to a customers location.
+
+* **Edge server**: An edge server is the same as an edge node.
+
+* **Origin**: An origin is an address (IP or domain) from which the CDN provider pulls content. A service can have multiple origins.
+
+* **Flavor**: A flavor is a configuration option. A flavor enables you to choose from a generic setting that is powered by one or more CDN providers.
```

Comment:
Just an FYI, the openstack poppy implementation does support other providers also (e.g. Fastly).  We are close to supporting all features with MaxCDN and Amazon CloudFront.  And EdgeCast have started building a provider too.